### PR TITLE
refactor: use 'untilTime' instead of 'maxLines' for logs pagination

### PR DESCRIPTION
**Pull request by alexmt**
Signed-off-by: Alexander Matyushentsev <AMatyushentsev@gmail.com>

Improves log viewer to use 'untilTime' instead of 'maxLines' for logs pagination.

### CODE REVIEW

File: ui/src/app/applications/components/application-details/application-details.scss [hunk 1]
```diff
@@ -43,7 +43,7 @@
     }
 
     &__tab-content-full-height {
-        height: calc(100vh - 2 * 50px);
+        height: calc(100vh - 2 * 76px);
```
Comment (alexmt) [inline ui/src/app/applications/components/application-details/application-details.scss:46]: Fixes scroll bug mentioned in https://github.com/argoproj/argo-cd/pull/5233#discussion_r562979174

File: ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx [hunk 2]
```diff
@@ -87,35 +95,41 @@ export const PodsLogsViewer = (props: {applicationName: string; pod: models.Reso
...
+                                    )
+                                    // show only current page lines
+                                    .scan((lines, logEntry) => {
+                                        lines.push(logEntry);
+                                        if (lines.length > maxLines) {
+                                            lines.splice(0, lines.length - maxLines);
+                                        }
+                                        return lines;
+                                    }, new Array<models.LogEntry>())
+                                    // re-render logs every 100ms
```
Comment (rbreeze) [inline ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx]: Why re-render if the `follow` option isn't set?

**Review by rbreeze — APPROVED** [review]
As long as there's a reason for re-rendering even when follow isn't selected, LGTM! Thanks!

File: ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx (hunk 2 above)
Comment (alexmt) [reply ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx]: Looks like comment is misleading. It should be `accumulate log changes and render only once every 100ms to reduce CPU usage`- fixed.

**Pull request merged**